include/ipc/ipc_service.h

```diff
@@ -70,20 +142,25 @@ struct ipc_ept_cfg {
 	void *priv;
 };
 
-/** @brief Register IPC endpoint.
+/** @brief Register IPC endpoint onto an instance.
  *
- *  Registers IPC endpoint to enable communication with a remote device.
+ *  Registers IPC endpoint onto an instance to enable communication with a
+ *  remote device.
  *
  *  The same function registers endpoints for both master and slave devices.
  *
  *  @param ept Endpoint object.
  *  @param cfg Endpoint configuration.
+ *  @param target_instance Name of the instance to register the endpoint onto.
+ *                         Can be NULL when the backend implements an internal
+ *                         logic for endpoint-instance allocation.
```

Comment:
Also null for backends that manage mono instance.proposal: 
Can be NULL depending on  the backend implementation ( default endpoint, or internal  logic for endpoint-instance allocation).

Reply:
ACK. I made the several cases explicit in the comment.